### PR TITLE
stdenv: drop remove unnecessary env var

### PR DESCRIPTION
**Pull request by Artturin**
and the associated obsolete functions

support for log nesting was removed in 2017 6669a3b

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by trofi — APPROVED** [review]

**Review by SuperSandro2000 — COMMENTED** [review]
great finding

File: pkgs/stdenv/generic/setup.sh [hunk 1]
```diff
@@ -1512,14 +1502,14 @@ showPhaseFooter() {
     local startTime="$2"
     local endTime="$3"
     local delta=$(( endTime - startTime ))
-    (( $delta < 30 )) && return
+    (( delta < 30 )) && return
```
Comment (Mindavi) [inline pkgs/stdenv/generic/setup.sh:1505]: Is the $ unneeded in math expressions?

**Review by Mindavi — APPROVED** [review]
Diff LGTM other than the one question

File: pkgs/stdenv/generic/setup.sh (hunk 1 above)
Comment (Artturin) [reply pkgs/stdenv/generic/setup.sh:1505]: Yeah 

SC2004 (style): $/${} is unnecessary on arithmetic variables.

File: pkgs/stdenv/generic/setup.sh [hunk 2]
```diff
@@ -125,11 +126,6 @@ _eval() {
 ######################################################################
 # Logging.
 
-# Obsolete.
-stopNest() { true; }
-header() { echo "$1"; }
-closeNest() { true; }
```
Comment (ncfavier) [inline pkgs/stdenv/generic/setup.sh:131]: A warning would have been nice before dropping those.

Comment (mweinelt) [reply pkgs/stdenv/generic/setup.sh:131]: I'm always surprised by this kind of out-of-tree usage and the expectation, that anyone in nixpkgs would know.

Comment (ncfavier) [reply pkgs/stdenv/generic/setup.sh:131]: Even without knowing, you can't assume that something is *not* used, if it's been exposed for some time (sadly). You never know how things bubble their way up from the entrails of nixpkgs to cargo-culted code snippets.

Comment (trofi) [reply pkgs/stdenv/generic/setup.sh:131]: For my curiosity: how would you notice the warning?

Comment (ncfavier) [reply pkgs/stdenv/generic/setup.sh:131]: Hopefully by looking at build logs. The longer the warnings stay around, the more likely that is to happen.

Comment (SuperSandro2000) [reply pkgs/stdenv/generic/setup.sh:131]: I have high doubts that those warnings inside of the build logs would have been noticed.

**Pull request merged**